src/libsystemd-network/sd-ipv4ll.c

```diff
@@ -257,7 +257,9 @@ static int ipv4ll_start_internal(sd_ipv4ll *ll, bool reset_generation) {
 
 int sd_ipv4ll_start(sd_ipv4ll *ll) {
         assert_return(ll, -EINVAL);
-        assert_return(sd_ipv4ll_is_running(ll) == 0, -EBUSY);
+
+        if (sd_ipv4ll_is_running(ll))
+                return 0;
 
         return ipv4ll_start_internal(ll, true);
```

Comment:
i think this makes sense, but I think this could use one more tweak: if we are already started we should return 1 here, but if we are just starting this we should return 0 here. i.e. two different ways of indicating success but still slightly different, to let the caller know if what they did was a NOP or not. This is in fact also hat sd_ndisc_start() does.

Reply:
One clarification, it looks like sd_ndisc_start() return values are 0 == already started, and 1 == successful start...should sd_ipv4ll_start() change to that?  Or 0 == successful start and 1 == already started?

Reply:
A quick look in src/libsystemd-network/ seems to show:
sd_dhcp_client_start: start == 0, already started == -EBUSY
sd_dhcp6_client_start: start == 0, already started == -EBUSY
sd_ipv4acd_start: start == 0, already started == -EBUSY
sd_ipv4ll_start: start == 0, already started == -EBUSY
sd_lldp_start: start == 1, already started == 0
sd_ndisc_start: start == 1, already started == 0
sd_radv_start: start == 0, already started == 0

should they be unified to a consistent set of return codes?

Reply:
of course, it would be excellent if this would be unified!

but i think I would like to change my mind regarding what i wrote above. I think == 0 should mean "already started, all ok", and == 1 (or more generically > 0) should mean "started now, all ok". i.e. sd_nids_start + sd_lldp_start do it right, and the others should be changed to work the same.

i.e. our usually rule regarding return values is < 0 error, >= 0 is success. And of the latter we sometimes define a special case > 0 meaning "we actually did something to become successful" while == 0 means "we didn't do anything, but things are OK anyway".

This is followed quite often in our codebase, but not always (as we can see), but I think we should strive to make this behaviour systematic.

(Doesn't have to be in this PR, but it would of course be excellent if this unification would happen, either here or in another PR)

Reply:
i adjusted sd_ipv4ll_start() to match, and i'll look at changing the others as well, but in a separate PR so it doesn't hold this one up (and in case it's not simple).  Thanks!